otelcol/command.go

```diff
@@ -52,13 +38,24 @@ func commandHelper(set CollectorSettings, enforceProviders bool) *cobra.Command
 		},
 	}
 	rootCmd.AddCommand(newComponentsCommand(set))
-	rootCmd.AddCommand(newValidateSubCommand(set, flagSet, enforceProviders))
+	rootCmd.AddCommand(newValidateSubCommand(set, flagSet))
 	rootCmd.Flags().AddGoFlagSet(flagSet)
 	return rootCmd
 }
 
+// NewCommandMustSetProvider constructs a new cobra.Command using the given CollectorSettings.
+// Any URIs specified in CollectorSettings.ConfigProviderSettings.ResolverSettings.URIs
+// are considered defaults and will be overwritten by config flags passed as
+// command-line arguments to the executable.
+// At least one Provider must be supplied via CollectorSettings.ConfigProviderSettings.ResolverSettings.ProviderFactories.
+//
+// Deprecated: [v0.104.0] use NewCommand instead
```

Comment:
We should probably have a deprecation changelog entry for this, as inconsequential as it likely is.